packages/nodes-base/nodes/Google/Sheet/test/v2/utils/utils.test.ts

```diff
@@ -0,0 +1,284 @@
+
+		expect(result).toBeDefined();
+		expect(result).toEqual([
+			['id', 'col1', 'col2', 'col3'],
+			[0, 'A', 'B', 'C'],
+		]);
+	});
+	it('should trimToFirstEmptyRow with row numbers', () => {
+		const data = [
+			['row_number', 'id', 'col1', 'col2', 'col3'],
+			[2, 0, 'A', 'B', 'C'],
+			[3, 1, 'a', 'b', 'c'],
+			[4, '', '', '', ''],
+			[5, 3, 'g', 'h', 'i'],
+		];
+
+		const result = trimToFirstEmptyRow(data);
+
+		expect(result).toBeDefined();
+		expect(result).toEqual([
+			['row_number', 'id', 'col1', 'col2', 'col3'],
+			[2, 0, 'A', 'B', 'C'],
+			[3, 1, 'a', 'b', 'c'],
+		]);
+	});
+});
+
+describe('Test Google Sheets, removeEmptyRows', () => {
+	it('should removeEmptyRows without row numbers', () => {
+		const data = [
+			['id', 'col1', 'col2', 'col3'],
+			[0, 'A', 'B', 'C'],
+			['', '', '', ''],
+			[2, 'd', 'e', 'f'],
+			['', '', '', ''],
+		];
+
+		const result = removeEmptyRows(data, false);
+
+		expect(result).toBeDefined();
+		expect(result).toEqual([
+			['id', 'col1', 'col2', 'col3'],
+			[0, 'A', 'B', 'C'],
+			[2, 'd', 'e', 'f'],
+		]);
+	});
+	it('should removeEmptyRows with row numbers', () => {
+		const data = [
+			['row_number', 'id', 'col1', 'col2', 'col3'],
+			[2, 0, 'A', 'B', 'C'],
+			[3, 1, 'a', 'b', 'c'],
+			[4, '', '', '', ''],
+			[5, 3, 'g', 'h', 'i'],
+		];
+
+		const result = removeEmptyRows(data);
+
+		expect(result).toBeDefined();
+		expect(result).toEqual([
+			['row_number', 'id', 'col1', 'col2', 'col3'],
+			[2, 0, 'A', 'B', 'C'],
+			[3, 1, 'a', 'b', 'c'],
+			[5, 3, 'g', 'h', 'i'],
+		]);
+	});
+});
+
+describe('Test Google Sheets, trimLeadingEmptyRows', () => {
+	it('should trimLeadingEmptyRows without row numbers', () => {
+		const data = [
+			['', '', '', ''],
+			['', '', '', ''],
+			[2, 'd', 'e', 'f'],
+			['', '', '', ''],
+		];
+
+		const result = trimLeadingEmptyRows(data, false);
+
+		expect(result).toBeDefined();
+		expect(result).toEqual([
+			[2, 'd', 'e', 'f'],
+			['', '', '', ''],
+		]);
+	});
+	it('should trimLeadingEmptyRows with row numbers', () => {
+		const data = [
+			[1, '', '', '', ''],
+			['row_number', 'id', 'col1', 'col2', 'col3'],
+			[2, 0, 'A', 'B', 'C'],
+			[3, 1, 'a', 'b', 'c'],
+			[5, 3, 'g', 'h', 'i'],
+		];
+
+		const result = trimLeadingEmptyRows(data, true);
+
+		expect(result).toBeDefined();
+		expect(result).toEqual([
+			['row_number', 'id', 'col1', 'col2', 'col3'],
+			[2, 0, 'A', 'B', 'C'],
+			[3, 1, 'a', 'b', 'c'],
+			[5, 3, 'g', 'h', 'i'],
+		]);
+	});
+});
+
+describe('Test Google Sheets, removeEmptyColumns', () => {
+	it('should removeEmptyColumns without row numbers', () => {
+		const data = [
+			['id', 'col1', '', 'col3'],
+			[0, 'A', '', 'C'],
+			[1, 'a', '', 'c'],
+			[2, 'd', '', 'f'],
+			[3, 'g', '', 'i'],
+		];
+
+		const result = removeEmptyColumns(data);
+
+		expect(result).toBeDefined();
+		expect(result).toEqual([
+			['id', 'col1', 'col3'],
+			['', 'A', 'C'], // TODO:should be [0, 'A', 'C'] ?
```

Comment:
Shouldn't we fix this? The problem seems to be [here](https://github.com/n8n-io/n8n/blob/18efaf397a6bab8bd5dba881bbdfeceef8dbafb0/packages/nodes-base/nodes/Google/Sheet/v2/helpers/GoogleSheets.utils.ts#L146).